### PR TITLE
Add examples for `unnecessary_lambda_linter()`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Haven't added examples for `map` family on account of #1871.

### CODE REVIEW

File: R/unnecessary_lambda_linter.R [hunk 1]
```diff
@@ -4,6 +4,19 @@
 #'   e.g. `lapply(DF, sum)` is the same as `lapply(DF, function(x) sum(x))` and
 #'   the former is more readable.
 #'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "lapply(list(1:3, 2:4), function(xi) { sum(xi) })",
```
Comment (MichaelChirico) [inline R/unnecessary_lambda_linter.R:10]: this violates `brace_linter()`, is that intentional?

**Comment by IndrajeetPatil** [comment]
Builds are failing due to https://github.com/r-lib/actions/issues/697

File: R/unnecessary_lambda_linter.R [hunk 2]
```diff
@@ -4,6 +4,19 @@
...
 #'   the former is more readable.
 #'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "lapply(list(1:3, 2:4), function(xi) sum(xi))",
+#'   linters = unnecessary_lambda_linter()
+#' )
+#'
+#' # okay
```
Comment (MichaelChirico) [inline R/unnecessary_lambda_linter.R:14]: two more OK examples to add:

`lapply(x, \(xi) grep("ptn", xi)` [i.e., where the arg doesn't come first]
`lapply(x, \(xi) data.frame(col = xi))` [i.e., where the argument is named

**Review by MichaelChirico — APPROVED** [review]
Thanks!

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**